File: project-euler/go/problem-002.go
package main

import "fmt"

func not(n int) int {
	if n == 0 {
		return 1
	}
	return 0
}

func solution() int {
	a1, a2, a3, sum := 0, 1, 2, 0
	for a3 < 4e6 {
		a3 = a1 + a2
		sum += a3 * not(a3%2)
		a1 = a2
		a2 = a3
	}
	return sum
}

func main() {
	fmt.Printf("%d\n", solution())
}

File: project-euler/ocaml/problem-002.ml
let solution =
  let rec aux acc x y =
    if x > 4_000_000 then acc
    else
      let y' = x + y and x' = y in
      aux (if x mod 2 = 0 then acc + x else acc) x' y'
  in
  aux 0 1 2
;;

print_int solution
